index.js

```diff
@@ -120,27 +149,85 @@ function setPolicy (s3, bucket, cb) {
   s3.getBucketPolicy({ Bucket: bucket }, function(err, data) {
     if (err && err.code !== 'NoSuchBucketPolicy') return cb(err)
 
-    var policy = {
+    var newPolicy = {
       Statement: []
     }
 
+    var oPolicy
+
     try {
-      policy = JSON.parse(data.Policy)
+      oPolicy = JSON.parse(data.Policy)
     } catch (err) {}
 
     var found = false
 
-    policy.Statement = policy.Statement.map(function(item) {
-      if (item.Sid === 'AddPublicReadPermissions') {
-        found = true
-        return publicRead
-      } else {
-        return item
-      }
+    if (oPolicy) {
+      newPolicy.Statement = oPolicy.Statement.map(function(item) {
+        if (item.Sid === 'AddPublicReadPermissions') {
+          found = true
+          return publicRead
+        } else {
+          return item
+        }
+      })
+    }
+
+    if (!found) newPolicy.Statement.push(publicRead)
+
+    var dirty = diff(oPolicy || {}, newPolicy, function(path, key) {
+      if (key === 'Version') return true
     })
 
-    if (!found) policy.Statement.push(publicRead)
+    if (dirty) {
+      var policy = assign(oPolicy || {}, newPolicy)
+      s3.putBucketPolicy({ Bucket: bucket, Policy: JSON.stringify(policy) }, cb)
+    } else {
+      cb()
+    }
+  })
+}
+
+function loadRoutes(routesOrFile) {
+  var routes
+  if (typeof routesOrFile === 'string') {
+    routes = require(path.resolve(__dirname, routesOrFile))
+  } else {
+    routes = routesOrFile
+  }
+
+  validateRoutes(routes)
+  return routes
+}
+
+function validateRoutes(routes) {
+  assert(routes.constructor === Array)
```

Comment:
If you just want to check if `routes` is an Array, I'd prefer `Array.isArray(routes)`